### DISCUSSION TITLE
Help calculating covariance matrices for FE54 with ENDFVIII data

### DISCUSSION THREAD

**Discussion by sombrereau** [njoy/NJOY21#149] [Q&A]
NJOY team,

I'm a very new NJOY user. I'm trying to generate the covariance data with NJOY to calculate the nuclear data uncertainty for a potential Plutonium Fueled Experimental Nuclear Reactor.

I'm getting errors when I try to process the ENDFVIII Fe54 nuclear data. NJOY creates tapes with empty covariance matrices for the MF33 table..

NJOY raises a segment fault message related to this issue in the errorr module

> zsh: segmentation fault  /Users/Cisneros/codes/NJOY21/bin/njoy21 -i

I'm running NJOY21 v. 1.1.1. I'm using nuclear data downloaded from the IAEA website (https://www-nds.iaea.org/public/download-endf/ENDF-B-VIII.0/n/n_2625_26-Fe-54.zip | copied as tape20) and I'm using the attached [input file](https://github.com/njoy/NJOY21/files/6717012/endf8_covariance_FE54_covariance.txt)

Thanks for any tips,

-Tommy Cisneros

**Comment by nathangibson14** [comment]
Hi Tommy,

Thanks for raising this issue.  Unfortunately, I don't have a great answer for you, and it might be some time before I do.

I recently attempted to process all of ENDF/B-VIII.0's MF33 through ERRORR, and Fe-54 was the only one that I was unable to obtain covariance matrices from NJOY for.  A few others gave me trouble sometimes but not always (e.g., group structure dependence), although I haven't dug into that and that might be a red herring.   But for sure, I agree something is up with Fe-54.

But that said, I'm not sure exactly what the problem is.  I'll be getting back to looking at that rather soon, and I'll update if I find anything.  Fe-54 is usually on the "problem nuclide" list because it's an LRF7 nuclide, but it doesn't have MF32 data, so this should be irrelevant here.

--Nathan

**Comment by sombrereau** [comment]
Hey @nathangibson14, thanks for looking into this.

**Comment by nathangibson14** [comment]
Hi @sombrereau,

I finally figured it out!

ERRORR checks if MF2 is LRF7 before it checks if MF32 is present.  Because Fe-54 is LRF7, it then moves into the MF32 processing for LRF7, and this leads to problems with MF32 not being present.  I'll get a branch of NJOY with this fix up within a day or two -- just need to make sure I didn't break anything else.

Fe-54 also has another problem.  MT851 is used for lumped covariance information, but the definition of the lump isn't given.  NJOY will crash because of this, and this is an evaluation problem, not an NJOY problem.  I'll consider whether NJOY should fail more gracefully, but the proper solution is to remove MF33/MT851 from the ENDF file until the evaluation is updated in a future release.

--Nathan
